File: exams/lomas/u3/Exam3PLomas/EXAM 3P/Exam/src/main/java/ec/edu/espe/exam/model/SortingContext.java
package ec.edu.espe.exam.model;

public class SortingContext {
    private SortingStrategy strategy;

    public int[] sort(int[] data) {
        int size = data.length;
        if (size >= 2 && size <= 6) {
            strategy = new BubbleSort();
        } else if (size >= 7 && size <= 10) {
            strategy = new InsertionSort();
        } else {
            strategy = new QuickSort();
        }
        return strategy.sort(data);
    }

    public String getAlgorithmName() {
        if (strategy != null) {
            return strategy.getClass().getSimpleName();
        }
        return "Unknown";
    }
}

File: exams/nacato/u3/ExamP3/StrategyJavaScript/sortingContext.js
const BubbleSort = require('./bubbleSort');
const InsertionSort = require('./insertionSort');
const QuickSort = require('./quickSort');

class SortingContext {
    constructor() {
        this.ss = null;
    }

    sort(data) {
        let size = data.length;
        this.ss = this.setSortStrategy(size);
        return this.ss.sort(data);
    }

    setSortStrategy(n) {
        if (n >= 2 && n <= 6) this.ss = new BubbleSort();
        else if (n >= 7 && n <= 10) this.ss = new InsertionSort();
        else if (n >= 11) this.ss = new QuickSort();
        return this.ss;
    }
}

module.exports = SortingContext;